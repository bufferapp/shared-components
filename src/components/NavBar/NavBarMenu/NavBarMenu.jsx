import React from 'react';
import PropTypes from 'prop-types';
import { ChevronDown } from '../../Icon';
import {
  NavBarStyled,
  NavBarEmail,
  NavBarName,
  NavBarUser,
  NavBarAvatar,
  NavBarChavron,
} from './style';

export function getUserAvatar(user) {
  if (user.avatar) {
    return user.avatar;
  }

  return 'https://s3.amazonaws.com/buffer-ui/Default+Avatar.png';
}

/** NavBar Menu component used by the Select component to show a custom User name and avatar
 *  button */
const NavBarMenu = ({ user, onClick }) => (
  <NavBarStyled onClick={onClick}>
    <NavBarUser>
      <NavBarName>{user.name}</NavBarName>
      <NavBarEmail>{user.email}</NavBarEmail>
    </NavBarUser>
    <NavBarAvatar avatar={getUserAvatar(user)} onClick={onClick} />
    <NavBarChavron>
      <ChevronDown color="grayLight" size="large" />
    </NavBarChavron>
  </NavBarStyled>
);

NavBarMenu.propTypes = {
  /** User Name and Email to be shown in the NavBar */
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,

  /** OnClick function to be called on Avatar click, passed by the Select component */
  onClick: PropTypes.func.isRequired,
};

export default NavBarMenu;
