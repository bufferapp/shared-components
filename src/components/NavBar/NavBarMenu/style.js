import styled from 'styled-components';
import {
  gray, grayDark, grayLight, white,
} from '../../style/colors';
import { fontSize, fontWeightMedium } from '../../style/fonts';

export const NavBarStyled = styled.div`
  display: flex;
  align-items: center;
  background: ${grayDark};
  padding: 12px 8px 12px 24px;
  cursor: pointer;
`;

export const NavBarUser = styled.div`
  display: flex;
  align-items: end;
  flex-direction: column;
  padding-right: 20px;
`;

export const NavBarAvatar = styled.div`
  border-radius: 100px;
  width: 40px;
  height: 40px;
  background: ${gray};
  background-size: cover;
  background-image: url(${props => (props.avatar)});
`;

export const NavBarName = styled.div`
  font-size: 16px;
  color: ${white};
  font-weight: ${fontWeightMedium};
`;

export const NavBarEmail = styled.div`
  font-size: ${fontSize};
  color: ${grayLight};
`;

export const NavBarChavron = styled.div`
  background: #3D3D3D;
  border: 1px solid #3D3D3D;
  box-sizing: border-box;
  border-radius: 4px;
  color: ${gray};
  height: 16px;
  width: 16px;
  display: flex;
  align-items: center;
  position: relative;
  left: -8px;
  cursor: pointer;
`;
