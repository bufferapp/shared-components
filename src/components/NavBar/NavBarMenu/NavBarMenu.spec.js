import snap from 'jest-auto-snapshots';
import NavBarMenu from './NavBarMenu';
import 'jest-styled-components';

snap(NavBarMenu, './NavBarMenu.jsx');
