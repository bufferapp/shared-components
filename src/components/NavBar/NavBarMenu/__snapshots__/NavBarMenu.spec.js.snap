// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > NavBarMenu Matches snapshot when passed only required props 1`] = `
<styled.div
  onClick={[MockFunction]}
>
  <styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
  </styled.div>
  <styled.div
    avatar="https://s3.amazonaws.com/buffer-ui/Default+Avatar.png"
    onClick={[MockFunction]}
  />
  <styled.div>
    <ChevronDownIcon
      color="grayLight"
      size="large"
    />
  </styled.div>
</styled.div>
`;
